### PR TITLE
feat(metrics): Add a metrics aggregator

### PR DESCRIPTION
**Pull request by jan-auer**
Adds an `Aggregator` actor which places metrics into buckets by time and tag value and flushes them out in regular time intervals to a recipient. Aggregators have their own internal lifecycle and will be instantiated per organization.

This PR also updates the protocol implementation. Instead of allowing any value with any metric type, the parser now requires `f64` for counters, distributions, and gauges. This will have sufficient accuracy for real-world use cases and allows for a fixed storage model. For sets, it allows arbitrary values and hashes them into a `u32`.

Follow-up to https://github.com/getsentry/relay/pull/948
Requires https://github.com/getsentry/relay/pull/957

### CODE REVIEW

**Comment by jan-auer** [comment]
Since the aggregator now requires a recipient for flushed buckets. To test this, we will need to merge #957 first, which moves the Actix test utilities into their own crate.

File: relay-metrics/src/bucketing.rs [hunk 1]
```diff
@@ -132,16 +133,24 @@ impl AggregatorConfig {
...
+                    initial_flush
+                } else {
+                    // If the initial flush time has passed, debounce future flushes with the
+                    // `debounce_delay` starting now.
+                    now + self.debounce_delay()
+                }
+            }
+            // Bucket from distant future. Best we can do is schedule it after initial delay
+            // TODO: or refuse metric altogether?
+            UnboundedInstant::Later => now + self.initial_delay(),
```
Comment (jjbayer) [inline relay-metrics/src/bucketing.rs]: @jan-auer Let me know if this makes sense. I'm unsure what to do about metrics with future timestamps.

Comment (jan-auer) [reply relay-metrics/src/bucketing.rs]: If the timestamp is not representable, then we're so far out of the valid ingestion range (something we've not yet implemented) that we can just drop the metric. WDYT if we just make this `Option<Instant>` everywhere and then bail out with an error to keep it simple?

Comment (jjbayer) [reply relay-metrics/src/bucketing.rs]: I think we need the ternary enum to support backdated metrics - at least if we want the tests to pass on macOS, where even the following panics:

```rust
use std::time::{Duration, Instant};

fn main() {
    let now = Instant::now();
    let then = now - Duration::from_secs(24 * 60 * 60);
    println!("{:?}", then);
}
```

**Review by jjbayer — APPROVED** [review]

**Pull request merged**